app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java

```diff
@@ -150,6 +161,42 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     @MainThread
     private void handleSignInResponse(int resultCode, Intent data) {
         if (resultCode == RESULT_OK) {
+            final String prevUid = data.getStringExtra(ExtraConstants.EXTRA_MERGE_FAILED);
+            if (prevUid != null) {
+                FirebaseDatabase.getInstance()
```

Comment:
Ok that's fine, documentation comes last anyway.  Let's save that for another time.